### PR TITLE
fix: access MATERIALIZED_VIEW with read_gbq

### PR DESCRIPTION
**Pull request by Genesis929**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/session/_io/bigquery/read_gbq_table.py [hunk 1]
```diff
@@ -124,6 +125,17 @@ def validate_table(
     if table_ref.dataset_id.startswith("_"):
         return False
 
+    # Materialized views，does not support snapshot
+    if table_type == "MATERIALIZED_VIEW":
+        warnings.warn(
```
Comment (shobsi) [inline bigframes/session/_io/bigquery/read_gbq_table.py:130]: we should add a test to test this warning

Comment (Genesis929) [reply bigframes/session/_io/bigquery/read_gbq_table.py:130]: Updated.

**Review by shobsi — APPROVED** [review]

File: tests/system/small/test_session.py [hunk 2]
```diff
@@ -390,9 +390,16 @@ def test_read_gbq_twice_with_same_timestamp(session, penguins_table_id):
     assert df3 is not None
 
 
-def test_read_gbq_on_linked_dataset_warns(session):
+@pytest.mark.parametrize(
+    "source_table",
+    [
+        "bigframes-dev.thelook_ecommerce.orders",
+        "bigframes-dev.bigframes_tests_sys.base_table_mat_view",
```
Comment (tswast) [inline tests/system/small/test_session.py:397]: I'm a concerned about hard coding this table. Could we at least include something in the `scripts/` directory that can recreate this table if needed?

Comment (Genesis929) [reply tests/system/small/test_session.py:397]: Got it, will add that.

**Pull request merged**